draft-ounsworth-pq-composite-sigs.mkd

```diff
@@ -357,6 +320,29 @@ _EDNOTE: This subsection will define the OIDs for the initial composite algorith
 # In Practice {#sec-in-pract}
 _EDNOTE: This section will talk about practical issue of how these certificates will be used. For example it will talk about the size of these certs and cert chains. It will explain that if a cert in the chain is a Composite cert then the whole chain needs to be of Composite Certs. It will also explain that the root CA cert does not have to be of the same algorithms. The root cert SHOULD NOT be transferred in the authentication exchange to save transport overhead and thus it can be different than the intermediate and leaf certs. It will talk about overhead (size and processing). It will also discuss backwards compatibility. It could include a subsection about implementation considerations._
 
+
+## PEM Storage of Composite Private Keys
+
+CompositePrivateKeys can be encoded to the PEM format by placing a ComposetiPrivateKey into the privateKey field of a PrivateKeyInfo or OneAsymmetricKey object, and then applying the PEM encoding rules as defined in [RFC7468] section 10 and 11 for plaintext and encrypted private keys, respectively.
+
+EDNOTE: Do we really need this? Isn't it obvious? We do want to open the can of worms of needing to add a similar sentence for every protocol and encoding under the sun?
```

Comment:
It should be obvious, and is reinforced by RFC 5958:
```
   -- An alternate representation that makes full use of ASN.1
   -- constraints follows.  Also note that PUBLIC-KEY needs to be
   -- imported from the new PKIX ASN.1 Algorithm Information module
   -- and PrivateKeyAlgorithms needs to be commented out.

   -- OneAsymmetricKey ::= SEQUENCE {
   --  version                  Version,
   --  privateKeyAlgorithm      SEQUENCE {
   --   algorithm                 PUBLIC-KEY.&id({PublicKeySet}),
   --    parameters               PUBLIC-KEY.&Params({PublicKeySet}
   --                               {@privateKeyAlgorithm.algorithm})
   --                               OPTIONAL}
   --  privateKey               OCTET STRING (CONTAINING
   --                             PUBLIC-KEY.&PrivateKey({PublicKeySet}
   --                             {@privateKeyAlgorithm.algorithm})),
   --  attributes           [0] Attributes OPTIONAL,
   --  ...,
   --  [[2: publicKey       [1] BIT STRING (CONTAINING
   --                             PUBLIC-KEY.&Params({PublicKeySet}
   --                             {@privateKeyAlgorithm.algorithm})
   --                             OPTIONAL,
   --  ...
   --    }
```
where the `privateKey`'s `PUBLIC-KEY.&PrivateKey` definition refers to the PRIVATE-KEY definition I added:
```
   pk-Composite PUBLIC-KEY ::= {
       IDENTIFIER id-Composite
       KEY CompositePublicKey
       PARAMS ARE absent
       CERT-KEY-USAGE
           { digitalSignature, nonRepudiation, keyCertSign, cRLSign }
       PRIVATE-KEY CompositePrivateKey
   }
```
where the `PRIVATE-KEY` and `.&PrivateKey` syntax are linked in the following definition in RFC 5912:
```
PUBLIC-KEY ::= CLASS {
    &id             OBJECT IDENTIFIER UNIQUE,
    &KeyValue       OPTIONAL,
    &Params         OPTIONAL,
    &paramPresence  ParamOptions DEFAULT absent,
    &keyUsage       KeyUsage OPTIONAL,
    &PrivateKey     OPTIONAL
} WITH SYNTAX {
    IDENTIFIER &id
    [KEY &KeyValue]
    [PARAMS [TYPE &Params] ARE &paramPresence]
    [CERT-KEY-USAGE &keyUsage]
    [PRIVATE-KEY &PrivateKey]
}
```

ASN.1 is confusing!!

Reply:
This is to say, I don't think we need that statement.

Reply:
Thanks Dan.

I'll leave this in for now. Let's discuss as a group on Thursday's call whether we want to rip out this section entirely.

Reply:
On the call we voted to leave it in to prevent someone from PEM encoding the composite key with mulitiple BEGIN / END blocks.